files/image_config/topology/topology.sh

```diff
@@ -7,15 +7,39 @@
 
 start() {
     TOPOLOGY_SCRIPT="topology.sh"
-    PLATFORM=${PLATFORM:-`sonic-cfggen -H -v DEVICE_METADATA.localhost.platform`}
-    HWSKU=${HWSKU:-`sonic-cfggen -d -v 'DEVICE_METADATA["localhost"]["hwsku"]'`}
+    PLATFORM=`sonic-cfggen -H -v DEVICE_METADATA.localhost.platform`
+    HWSKU=`sonic-cfggen -d -v 'DEVICE_METADATA["localhost"]["hwsku"]' 2>&1`
+    if [[ $? -ne 0 || HWSKU == "" ]]; then
+            if [[ -f "/etc/sonic/minigraph.xml" ]]; then
+                HWSKU=`sonic-cfggen -m /etc/sonic/minigraph.xml -v "DEVICE_METADATA['localhost']['hwsku']" 2>&1`
+                if [[ $? -ne 0 || HWSKU == "" ]]; then
+                    echo "Failed to get HWSKU"
+                    exit 1 
+                fi
+            else
+                echo "Failed to get HWSKU"
+                exit 1
+            fi
+    fi
     /usr/share/sonic/device/$PLATFORM/$HWSKU/$TOPOLOGY_SCRIPT start
 }
 stop() {
     TOPOLOGY_SCRIPT="topology.sh"
-    PLATFORM=${PLATFORM:-`sonic-cfggen -H -v DEVICE_METADATA.localhost.platform`}
-    HWSKU=${HWSKU:-`sonic-cfggen -d -v 'DEVICE_METADATA["localhost"]["hwsku"]'`}
-    usr/share/sonic/device/$PLATFORM/$HWSKU/$TOPOLOGY_SCRIPT stop 
+    PLATFORM=`sonic-cfggen -H -v DEVICE_METADATA.localhost.platform`
+    HWSKU=`sonic-cfggen -d -v 'DEVICE_METADATA["localhost"]["hwsku"]' 2>&1`
+    if [[ $? -ne 0 || HWSKU == "" ]]; then
+            if [[ -f "/etc/sonic/minigraph.xml" ]]; then
+                HWSKU=`sonic-cfggen -m /etc/sonic/minigraph.xml -v "DEVICE_METADATA['localhost']['hwsku']" 2>&1`
```

Comment:
we can add this logic in function instead of duplicating it for start/stop

Reply:
Updated as suggested.